### PR TITLE
Developed updateApplicationDraft database function

### PR DESCRIPTION
**Pull request by yashodgayashan**
Seems mongodb connector don't have much features such as $set, $push, $inc in updated operation.

### CODE REVIEW

File: core/src/env_platform/db_handler.bal [hunk 1]
```diff
@@ -93,11 +93,51 @@ function getApplicationStatusByApplicationId(string applicationId) returns strin
...
     if (application is map<json>) {
         return trap <string>application.status;
     } else {
         return application;
     }
 }
+
+# The `putDraftApplication` function will alter the exsisting application and add the incoming form.
+# 
+# + form - The TreeRemovalForm Type record is accepted.
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:106]:
```suggestion
# + form - Form containing the tree removal data.
```

File: core/src/env_platform/db_handler.bal [hunk 2]
```diff
@@ -93,11 +93,51 @@ function getApplicationStatusByApplicationId(string applicationId) returns strin
...
-
     map<json>|error application = trap find[0];
     if (application is map<json>) {
         return trap <string>application.status;
     } else {
         return application;
     }
 }
+
+# The `putDraftApplication` function will alter the exsisting application and add the incoming form.
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:104]:
```suggestion
# The `updateApplicationDraft` function will alter the exsisting application draft with the incoming form details.
```

File: core/src/env_platform/db_handler.bal [hunk 3]
```diff
@@ -93,11 +93,51 @@ function getApplicationStatusByApplicationId(string applicationId) returns strin
...
+# 
+# + form - The TreeRemovalForm Type record is accepted.
+# + return - This function will return true if draft is added to the database or else return mongodb:Database error.
+function putDraftApplication(TreeRemovalForm form, string applicationId) returns boolean|error {
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:108]:
```suggestion
function updateApplicationDraft(TreeRemovalForm form, string applicationId) returns boolean|error {
```

File: core/src/env_platform/db_handler.bal [hunk 4]
```diff
@@ -93,11 +93,51 @@ function getApplicationStatusByApplicationId(string applicationId) returns strin
...
+        "district": form.district,
+        "nameOfTheLand": form.nameOfTheLand,
+        "planNumber": form.planNumber,
+        "area": extractAreaAsJSONArray(form.area),
+        "treeInformation": extractTreeInformationAsJSONArray(form.treeInformation)
+    };
+
+    int|mongodb:DatabaseError update = applicationCollection->update({"versions.0": application}, {"applicationId": applicationId});
+
+    return update is mongodb:DatabaseError ? update : true;
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:142]:
```suggestion
    if (updated is int) {
        return updated == 1 ? true:false;
    } else {
        return updated;
    }
```

File: core/src/env_platform/db_handler.bal [hunk 5]
```diff
@@ -93,11 +93,51 @@ function getApplicationStatusByApplicationId(string applicationId) returns strin
...
+        "treeRemovalAuthority": form.treeRemovalAuthority,
+        "city": form.city,
+        "district": form.district,
+        "nameOfTheLand": form.nameOfTheLand,
+        "planNumber": form.planNumber,
+        "area": extractAreaAsJSONArray(form.area),
+        "treeInformation": extractTreeInformationAsJSONArray(form.treeInformation)
+    };
+
+    int|mongodb:DatabaseError update = applicationCollection->update({"versions.0": application}, {"applicationId": applicationId});
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:140]:
```suggestion
    int|mongodb:DatabaseError updated = applicationCollection->update({"versions.0": application}, {"applicationId": applicationId});
```

File: core/src/env_platform/db_handler.bal [hunk 6]
```diff
@@ -93,11 +93,51 @@ function getApplicationStatusByApplicationId(string applicationId) returns strin
...
         return trap <string>application.status;
     } else {
         return application;
     }
 }
+
+# The `putDraftApplication` function will alter the exsisting application and add the incoming form.
+# 
+# + form - The TreeRemovalForm Type record is accepted.
+# + return - This function will return true if draft is added to the database or else return mongodb:Database error.
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:107]:
```suggestion
# + return - This function will return true if draft is updated in the database, false if not or else it returns a mongodb:Database error.
```

**Comment by yashodgayashan** [comment]
@deshankoswatte completed requested changes by the way ballerina mongodb connector doesn't support $push, $inc kind of operations then it is bit difficult to do some tasks.

**Comment by deshankoswatte** [comment]
> @deshankoswatte completed requested changes by the way ballerina mongodb connector doesn't support $push, $inc kind of operations then it is bit difficult to do some tasks.

Yeah ballerina is still in the early stages with this connector may need to talk with Joe about it.

**Comment by yashodgayashan** [comment]
> > @deshankoswatte completed requested changes by the way ballerina mongodb connector doesn't support $push, $inc kind of operations then it is bit difficult to do some tasks.
> 
> Yeah ballerina is still in the early stages with this connector may need to talk with Joe about it.

@deshankoswatte  Yes I will try to do the database operation with existing operations and Could you please go through some resource functions when you are free.

**Comment by yashodgayashan** [comment]
@sanjulamadurapperuma any comments?

**Comment by sanjulamadurapperuma** [comment]
> @sanjulamadurapperuma any comments?

I'll check

**Pull request merged**